### ISSUE TITLE
Make Dropdown code reusable and reuse it among filters

### ISSUE THREAD

**Issue by kmanijak** [woocommerce/woocommerce-gutenberg-products-block#7875]
#### Description
There's couple of filters supporting dropdown version:
- [Filter by Attribute](https://github.com/woocommerce/woocommerce-blocks/tree/trunk/assets/js/blocks/attribute-filter)
- [Filter by Rating - PR](https://github.com/woocommerce/woocommerce-blocks/pull/7771)
- [Filter by Stock - PR](https://github.com/woocommerce/woocommerce-blocks/pull/7831)

All of them have pretty similar implementation and there's quite a lot of duplicated code, mechanisms or styles.

#### Goal
The goal of this issue is to extract common logic/styles (whatever applicable) and reuse it among blocks to reuse the code and decrease the bundle size.

**Comment by roykho** [comment]
So after taking a look at this, I am not sure if it warrants an extraction. There are indeed some repeated items however you would still need to pass many props such as `remountKey`, `placeholder`, `displayTransform`, `messages` and even `styles` if they're different between filter blocks.

If we did this, it would just mean we've replaced calling a component with another and not much gain. Unless I missed something, wouldn't mind discussing this further.

**Comment by kmanijak** [comment]
Even though I was the one who created the issue, looking at this now, I must agree with you - it wouldn't add much value, while it could be time-consuming.

Maybe (just a maybe) it would make more sense to abstract "Filter" mechanism, since there's a lot of in common between:
- Filter by Attribute
- Filter by Rating
- Filter by Stock
- Filter by Price (although this one differs from the rest).
But it will be pretty time-consuming as well and with a pretty complicated Filter by Attribute's logic it can be really troublesome to make a proper abstraction.
The gain would be: simplified code-base (the logic there is very complicated and can be simplified for sure), a bit smaller bundle-size (Filters are currently one of the biggest blocks in the WooCommerce blocks, comparable with Cart and Checkout).

To summarise:
- I agree with your point that it's pointless to abstract Dropdown logic
- I think we could repurpose issue to abstract Filters in general which would have good impact (especially would improve the DX), but it's time-consuming, so I'm not sure if we should invest time in it.

**Comment by roykho** [comment]
> abstract Filters in general 

Yeah I think that makes more sense.

**Comment by kmanijak** [comment]
There's a dedicated project covering the Frontend Filters. Closing this one.
